tests/unit/amo/pages/TestSearchPage.js

```diff
@@ -810,4 +850,907 @@ describe(__filename, () => {
+      const category = 'causes';
+      const categoryName = 'Causes';
+
+      _loadCategories();
+
+      _dispatchSearchResults({
+        filters: {
+          addonType: ADDON_TYPE_STATIC_THEME,
+          category,
+        },
+      });
+
+      render({ category, type: ADDON_TYPE_STATIC_THEME });
+
+      expect(
+        screen.getByRole('heading', {
+          name: `2 themes found in ${categoryName}`,
+        }),
+      ).toBeInTheDocument();
+    });
+
+    it('should render results without categoryName or query when neither are present for addonType ADDON_TYPE_STATIC_THEME', () => {
+      _dispatchSearchResults({
+        filters: {
+          addonType: ADDON_TYPE_STATIC_THEME,
+        },
+      });
+
+      render({ type: ADDON_TYPE_STATIC_THEME });
+
+      expect(
+        screen.getByRole('heading', {
+          name: '2 themes found',
+        }),
+      ).toBeInTheDocument();
+    });
+
+    it('should render singular form when only one result is found with addonType ADDON_TYPE_EXTENSION', () => {
+      const query = 'test';
+      _dispatchSearchResults({
+        addons: [fakeAddon],
+        filters: {
+          addonType: ADDON_TYPE_EXTENSION,
+          query,
+        },
+      });
+
+      render({ query, type: ADDON_TYPE_EXTENSION });
+
+      expect(
+        screen.getByRole('heading', {
+          name: `1 extension found for "${query}"`,
+        }),
+      ).toBeInTheDocument();
+    });
+
+    it('should render plural form when multiple results are found with addonType ADDON_TYPE_EXTENSION', () => {
+      const query = 'test';
+      _dispatchSearchResults({
+        filters: {
+          addonType: ADDON_TYPE_EXTENSION,
+          query,
+        },
+      });
+
+      render({ query, type: ADDON_TYPE_EXTENSION });
+
+      expect(
+        screen.getByRole('heading', {
+          name: `2 extensions found for "${query}"`,
+        }),
+      ).toBeInTheDocument();
+    });
+
+    it('should render results with categoryName and query for addonType ADDON_TYPE_EXTENSION when search is loaded', () => {
+      const query = 'test';
+      const category = 'bookmarks';
+      const categoryName = 'Bookmarks';
+
+      _loadCategories({
+        results: [
+          {
+            ...fakeCategory,
+            type: ADDON_TYPE_EXTENSION,
+            name: categoryName,
+            slug: category,
+          },
+        ],
+      });
+
+      _dispatchSearchResults({
+        filters: {
+          addonType: ADDON_TYPE_EXTENSION,
+          category,
+          query,
+        },
+      });
+
+      render({ category, query, type: ADDON_TYPE_EXTENSION });
+
+      expect(
+        screen.getByRole('heading', {
+          name: `2 extensions found for "${query}" in ${categoryName}`,
+        }),
+      ).toBeInTheDocument();
+    });
+
+    it('should render results with categoryName and no query for addonType ADDON_TYPE_EXTENSION when there is no query and when search is loaded', () => {
```

Comment:
I really just copied all of the tests from `TestSearchContextCard.js` and adjusted them to work, without looking for specific opportunities for improvements. I can take a look and see if this would be a relatively simple improvement to make.

Reply:
I did what you suggested. Because of the numbers of permutations and combinations, I ended up with a lot of code, but probably less than was there before, and, as you pointed out, it makes it easier to see which combinations are being tested.